ugs-core/src/com/willwinder/universalgcodesender/uielements/jog/StepSizeSpinnerModel.java

```diff
@@ -43,8 +40,11 @@ public Object getNextValue() {
                 stepSize = stepSize / 10;
             }
         }
-        setStepSize(stepSize);
-        return super.getNextValue();
+
+        if( value >= MAX_VALUE) {
+            return 0;
```

Comment:
This should probably be `return MAX_VALUE`, to prevent cycling the step size back to zero. And below should be `return MIN_VALUE`